### PR TITLE
nng v1.0.1 (new formula)

### PR DESCRIPTION
**Pull request by neachdainn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

A few questions about this, as this is my first pull request to this repository and my first time writing anything for Homebrew:

1. The formula passes `brew audit --strict nng` but I get errors from `brew audit --new-formula nng`, including errors that saying that the formula shouldn't have options. Should I really not have options in the formula? I think the ones presented are reasonable. It also says I shouldn't have a `HEAD` spec?
2. Is there a way to have Asciidoctor as a build dependency only if the user wants to build docs?
3. Is `--with-shared-library` the best name for the option to build nng as a shared library?

I am willing to iterate on this until it's perfect, I'm just not sure what perfect is in this context.

### CODE REVIEW

**Comment by lembacon** [comment]
We do not want options and HEAD for new formulae. Please see #31510.

**Comment by neachdainn** [comment]
Options and HEAD have been removed.

File: Formula/nng.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  depends_on "asciidoctor" => :build
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-GNinja", "-DNNG_ENABLE_DOC=ON", "-DBUILD_SHARED_LIBS=ON", *std_cmake_args
+      system "ninja"
+      system "ninja", "test" if build.with? "test"
```
Comment (fxcoudert) [inline Formula/nng.rb]: Remove this line.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @neachdainn for the pull request!

**Pull request closed without merging**